src/components/Todos/TodoList.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useContext, useEffect } from "react";
+import { useHistory } from "react-router";
+import axios from "axios";
+
+import classes from "./TodoList.module.css";
+import AuthContext from "../../store/AuthContext";
+
+const TodoList = () => {
+    const [todoList, setTodoList] = useState(null);
+    const [error, setError] = useState(null);
+    const authCtx = useContext(AuthContext);
+    const token = authCtx.token;
+    const history = useHistory();
+    const [isLoading, setIsLoading] = useState(null);
+
+    useEffect(() => {
+        setIsLoading(true);
+        axios
+            .get("http://localhost:5151/todos", {
+                headers: {
+                    Authorization: token,
+                },
+            })
+            .then((response) => {
+                setTodoList(response.data.data);
+                setError(null);
+                setIsLoading(false);
+            })
+            .catch((error) => {
+                if (error.response) {
+                    if (error.response.status === 401) {
+                        history.push("/login");
+                        return;
+                    }
+                }
+                setError(error.message);
+            });
```

Comment:
As my  comment for last time, you should hide loading when request is completed. Not only success, so that, you need hide loading after .catch, please check below syntax carefully

```js
axios.getOrPost('/v1/api_endpoint')
  .then((res) => { // handle success })
  .catch((err) => { // handle error })
  .then(() => { // always executed, you should hide loading here })  
```

Reply:
```js
axios
            .get("http://localhost:5151/todos", {
                headers: {
                    Authorization: token,
                },
            })
            .then((response) => {
                setTodoList(response.data.data);
                setError(null);
            })
            .catch((error) => {
                if (error.response) {
                    if (error.response.status === 401) {
                        history.push("/login");
                        return;
                    }
                }
                setError(error.message);
            });
           .then(() => { 
             setIsLoading(false);
           })